### ISSUE TITLE
NETStandard.Library 2.0.2-servicing-25519-03 was not found

### ISSUE THREAD

**Issue by magick93** [redhat-developer/s2i-dotnetcore#221]
Hello

I'm getting a build error - and even with `--loglevel=5 ` I'm still not getting any useful info from the output.

This is what I get.

```
I0910 18:15:55.940469   10632 docker.go:1055] Waiting for container "4385324651a0e1b5a809cd43ecf93044917565b0e15e6452aa612ce297e1d4bb" to stop ...
I0910 18:15:56.172918   10632 docker.go:991] Removing container "4385324651a0e1b5a809cd43ecf93044917565b0e15e6452aa612ce297e1d4bb" ...
I0910 18:15:56.603737   10632 docker.go:1001] Removed container "4385324651a0e1b5a809cd43ecf93044917565b0e15e6452aa612ce297e1d4bb"
I0910 18:15:56.604041   10632 cleanup.go:33] Removing temporary directory /tmp/s2i472911301
I0910 18:15:56.604059   10632 fs.go:278] Removing directory '/tmp/s2i472911301'
I0910 18:15:56.694340   10632 build.go:171] Build failed
E0910 18:15:56.694368   10632 errors.go:276] An error occurred: non-zero (13) exit code from myrepo:5000/dotnet-21-ignite-centos7
```

Any help is greatly appreciated.

**Comment by tmds** [comment]
This log doesn't show if the issue is related to the .NET Core builder. Do you have the build log from what happens in the build pod (this one starts with the git commit being checked out)?

**Comment by magick93** [comment]
I dont have access to the logs now - as I'm at home and theyre only accessible from the office.

However, I do know there was nothing out of the ordinary in the builder output. It appeared to build as expected.

**Comment by magick93** [comment]
Ok, I found the problem:

```
I0911 08:02:45.405323   16262 sti.go:681] Restore completed in 4.05 sec for /opt/app-root/src/MyCo.Services.Product.CacheEntity/MyCo.Services.Product.CacheEntity.csproj.
I0911 08:02:45.490486   16262 sti.go:681] /opt/app-root/src/MyCo.Infrastructure.Ignite/MyCo.Infrastructure.Ignite.csproj : error NU1603: MyCo.Infrastructure.Ignite depends on NETStandard.Library (>= 2.0.2-servicing-25519-03) but NETStandard.Library 2.0.2-servicing-25519-03 was not found. An approximate best match of NETStandard.Library 2.0.2 was resolved. [/opt/app-root/src/MyCo.Exchange.Api/MyCo.Exchange.Api.csproj]
I0911 08:02:45.533564   16262 sti.go:681] /opt/app-root/src/MyCo.HealthChecks/MyCo.HealthChecks.csproj : error NU1603: MyCo.HealthChecks depends on NETStandard.Library (>= 2.0.2-servicing-25519-03) but NETStandard.Library 2.0.2-servicing-25519-03 was not found. An approximate best match of NETStandard.Library 2.0.2 was resolved. [/opt/app-root/src/MyCo.Exchange.Api/MyCo.Exchange.Api.csproj]
```
Basically it seems there is a mis-match in the minor version of the dotnetcore framework.

Is there any guidance on how to upgrade this?

**Comment by tmds** [comment]
You libraries refer to NETStandard.Library versions which aren't on nuget.org: https://www.nuget.org/packages/NETStandard.Library/.
You should specify the TargetFramework in your library csproj files (e.g. netstandard2.0) and not add explicit references to NETStandard.Library. The sdk will pick the proper versions for you.

**Comment by tmds** [comment]
I've looked into this further. This is an issue in our sdk, I've created a bugzilla ticket: https://bugzilla.redhat.com/show_bug.cgi?id=1627973

You should be able to work around this by disable the TreatWarningsAsErrors or adding an explicit package reference to NETStandard.Library in your library csproj files.

**Comment by tmds** [comment]
For example:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="NETStandard.Library" Version="2.0.3" />
  </ItemGroup>

</Project>
```

I'll post an update in this issue when we fixed the sdk.

**Comment by tmds** [comment]
This is fixed in the 2.1.402 sdk. You can use that sdk by setting DOTNET_SDK_VERSION or adding a global.json file to your repo.
